### PR TITLE
update tendermint.ts class

### PR DESCRIPTION
**Pull request by Seb-figment**

### CODE REVIEW

File: chain/tendermint.ts [hunk 1]
```diff
@@ -1,7 +1,334 @@
 import '../common/eager_offset'
 import { Bytes } from '../common/collections'
-
```
Comment (pienkowb) [inline chain/tendermint.ts:3]: Let's keep this line.

File: chain/tendermint/proto/package.json [hunk 2]
```diff
@@ -7,15 +7,20 @@
     "scripts": {
         "tendermint": "protoc --plugin=protoc-gen-as=./node_modules/.bin/as-proto-gen --proto_path=./src --as_out=../ ./src/t.v1.proto && mv ../t.v1.ts ../t.ts && FILE_PATH=../t.ts node index.js",
         "cosmos": "protoc --plugin=protoc-gen-as=./node_modules/.bin/as-proto-gen --proto_path=./src --as_out=../ ./src/cosmos.v1.proto && mv ../cosmos.v1.ts ../cosmos.ts && FILE_PATH=../cosmos.ts node index.js",
-        "liquidity": "protoc --plugin=protoc-gen-as=./node_modules/.bin/as-proto-gen --proto_path=./src --as_out=../ ./src/liquidity.v1.proto && mv ../liquidity.v1.ts ../liquidity.ts && FILE_PATH=../liquidity.ts node index.js"
+        "liquidity": "protoc --plugin=protoc-gen-as=./node_modules/.bin/as-proto-gen --proto_path=./src --as_out=../ ./src/liquidity.v1.proto && mv ../liquidity.v1.ts ../liquidity.ts && FILE_PATH=../liquidity.ts node index.js",
```
Comment (pienkowb) [inline chain/tendermint/proto/package.json:10]: Any reason this is not included in `cosmos`?

File: chain/tendermint/proto/package.json [hunk 3]
```diff
@@ -7,15 +7,20 @@
     "scripts": {
         "tendermint": "protoc --plugin=protoc-gen-as=./node_modules/.bin/as-proto-gen --proto_path=./src --as_out=../ ./src/t.v1.proto && mv ../t.v1.ts ../t.ts && FILE_PATH=../t.ts node index.js",
         "cosmos": "protoc --plugin=protoc-gen-as=./node_modules/.bin/as-proto-gen --proto_path=./src --as_out=../ ./src/cosmos.v1.proto && mv ../cosmos.v1.ts ../cosmos.ts && FILE_PATH=../cosmos.ts node index.js",
```
Comment (pienkowb) [inline chain/tendermint/proto/package.json:9]: This is outside of the scope of this task but we need to think of a better name for this – we'll have a name clash after changing `tendermint` to `cosmos`.

File: test/test.js [hunk 4]
```diff
@@ -32,6 +36,15 @@ async function main() {
   fs.copyFileSync('chain/ethereum.ts', 'test/temp_lib/chain/ethereum.ts')
   fs.copyFileSync('chain/near.ts', 'test/temp_lib/chain/near.ts')
   fs.copyFileSync('chain/tendermint.ts', 'test/temp_lib/chain/tendermint.ts')
+  fs.copyFileSync('chain/tendermint/t.ts', 'test/temp_lib/chain/tendermint/t.ts')
+  fs.copyFileSync(
+    'chain/tendermint/liquidity.ts',
+    'test/temp_lib/chain/tendermint/liquidity.ts',
+  )
```
Comment (pienkowb) [inline test/test.js:43]: Do we need `liquidity` as a separate file?

File: chain/tendermint/proto/package.json (hunk 2 above)
Comment (Seb-figment) [reply chain/tendermint/proto/package.json:10]: @patiee ?

File: chain/tendermint/proto/package.json [hunk 5]
```diff
@@ -7,15 +7,20 @@
...
-        "liquidity": "protoc --plugin=protoc-gen-as=./node_modules/.bin/as-proto-gen --proto_path=./src --as_out=../ ./src/liquidity.v1.proto && mv ../liquidity.v1.ts ../liquidity.ts && FILE_PATH=../liquidity.ts node index.js"
+        "liquidity": "protoc --plugin=protoc-gen-as=./node_modules/.bin/as-proto-gen --proto_path=./src --as_out=../ ./src/liquidity.v1.proto && mv ../liquidity.v1.ts ../liquidity.ts && FILE_PATH=../liquidity.ts node index.js",
+        "lint": "prettier -c ../*.ts",
+        "format": "prettier --write -c ../*.ts"
```
[the remark this answers is not in the document]
Comment (Seb-figment) [reply chain/tendermint/proto/package.json:12]: The top level script doesn't catch them.

File: test/test.js [hunk 6]
```diff
@@ -32,6 +36,15 @@ async function main() {
   fs.copyFileSync('chain/ethereum.ts', 'test/temp_lib/chain/ethereum.ts')
   fs.copyFileSync('chain/near.ts', 'test/temp_lib/chain/near.ts')
   fs.copyFileSync('chain/tendermint.ts', 'test/temp_lib/chain/tendermint.ts')
+  fs.copyFileSync('chain/tendermint/t.ts', 'test/temp_lib/chain/tendermint/t.ts')
```
[the remark this answers is not in the document]
Comment (Seb-figment) [reply test/test.js:39]: I suggest we handle this in the renaming task

File: test/test.js (hunk 4 above)
Comment (Seb-figment) [reply test/test.js:43]: @patiee ?

File: chain/tendermint.ts [hunk 7]
```diff
@@ -2,20 +2,6 @@ import '../common/eager_offset'
...
-    SIGNED_MSG_TYPE_PROPOSAL = 32,
-  }
-
-  export enum BlockIDFlag {
-    BLOCK_ID_FLAG_UNKNOWN = 0,
-    BLOCK_ID_FLAG_ABSENT = 1,
-    BLOCK_ID_FLAG_COMMIT = 2,
-    BLOCK_ID_FLAG_NIL = 3,
-  }
-
```
Comment (pienkowb) [inline chain/tendermint.ts:18]: I think we should keep enums at the top (similarly to `codec.proto`).

**Review by pienkowb — APPROVED** [review]

**Pull request merged**